### PR TITLE
Instrument basic metrics within the compositor

### PR DESCRIPTION
**Pull request by chinmaygarde**

### CODE REVIEW

File: sky/compositor/instrumentation.h [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+
+  void reset(size_t count = 0) { _count = count; }
+
+  void increment(size_t count = 1) { _count += count; }
+
+ private:
+  size_t _count;
+
+  DISALLOW_COPY_AND_ASSIGN(Counter);
+};
```
Comment (abarth) [inline sky/compositor/instrumentation.h]: Why is a Counter better than just a size_t ?

**Comment by abarth** [comment]
LGTM, but I'd ditch Counter and just use a size_t or an int

File: sky/compositor/instrumentation.h (hunk 1 above)
Comment (chinmaygarde) [inline sky/compositor/instrumentation.h]: The idea is that certain counters that measure frame specific (or other "instantaneous") information  are useless on their own and more insights can be gleaned from graphing out how these values change over time. For such cases, the past `n` counts could be recorded (on `reset`) and charted. So this class is more of a stub.

Comment (abarth) [inline sky/compositor/instrumentation.h]: ok

**Pull request merged**